integration/hostuser_test.go

```diff
@@ -431,6 +454,66 @@ func TestRootHostUsers(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("Test default shell assignment", func(t *testing.T) {
+		defaultShellUser := "default-shell"
+		namedShellUser := "named-shell"
+		absoluteShellUser := "absolute-shell"
+
+		t.Cleanup(func() { cleanupUsersAndGroups([]string{defaultShellUser, namedShellUser, absoluteShellUser}, nil) })
+
+		// Create a user with a named shell expected to be available in the PATH
+		users := srv.NewHostUsers(context.Background(), presence, "host_uuid")
+		_, err := users.UpsertUser(namedShellUser, services.HostUsersInfo{
+			Mode:  services.HostUserModeKeep,
+			Shell: "bash",
+		})
+		require.NoError(t, err)
+
+		// Create a user with an absolute path to a shell
+		_, err = users.UpsertUser(absoluteShellUser, services.HostUsersInfo{
+			Mode:  services.HostUserModeKeep,
+			Shell: "/usr/bin/bash",
+		})
+		require.NoError(t, err)
+
+		// Create a user with the host default shell (default behavior)
+		_, err = users.UpsertUser(defaultShellUser, services.HostUsersInfo{
+			Mode: services.HostUserModeKeep,
+		})
+		require.NoError(t, err)
+
+		_, err = user.Lookup(namedShellUser)
+		require.NoError(t, err)
+
+		_, err = user.Lookup(absoluteShellUser)
+		require.NoError(t, err)
+
+		_, err = user.Lookup(defaultShellUser)
+		require.NoError(t, err)
+
+		// Verify users have the correct shell assigned
+		userShells, err := getUserShells("/etc/passwd")
+		require.NoError(t, err)
```

Comment:
Yeah I considered this too 🤔 Would there be any benefits to doing one versus the other? Parsing the file seemed simpler to me than spinning up commands, but I'd be happy to swap for this if it's a more robust test

Reply:
> I don't think there are any other tests that both set a custom shell and start an SSH session.

There aren't, but I can take a look at adding one if you think we need the extra coverage.  Although that seems like testing the behavior of the host moreso than the behavior of Teleport since how things are handled after updating `/etc/passwd` is up to the host